src/sst/core/model/python/pymodel_comp.cc

```diff
@@ -271,6 +268,63 @@ static PyObject* compSetSubComponent(PyObject *self, PyObject *args)
     return nullptr;
 }
 
+/**
+static PyObject* compReuseStatistic(PyObject *self, PyObject *args)
+{
+    char *name = nullptr;
+
+    if ( !PyArg_ParseTuple(args, "0", &statoObj) )
+        return nullptr;
+
+    c->addStatistic(statObj->configStatistic->id, name);
+
+
+    if ( !PyArg_ParseTuple(args, "s", &name) )
+        return nullptr;
+
+    ConfigComponent *c = getComp(self);
+    if ( nullptr == c ) return nullptr;
+
+    StatisticId_t stat_id = c->getNextStatisticID();
+    SST::Experimental::ConfigStatistic* stat = c->addStatistic(stat_id, name);
+    if ( nullptr != stat ) {
+        PyObject *argList = Py_BuildValue("Ok", self, stat_id);
+        PyObject *statObj = PyObject_CallObject((PyObject*)&Experimental::PyModel_StatType, argList);
+        Py_DECREF(argList);
+        return statObj;
+    }
+
+    char errMsg[1024] = {0};
+    snprintf(errMsg, sizeof(errMsg)-1, "Failed to create statistic %s on %s. Already attached a statistic with that name ?\n", name, c->name.c_str());
+    PyErr_SetString(PyExc_RuntimeError, errMsg);
+    return nullptr;
+}
+*/
+
+static PyObject* compSetStatistic(PyObject *self, PyObject *args)
```

Comment:
We need the assign statistic function to re-use an existing stat object

Reply:
setStatistic seems to repeat most, if not all, of the code in createStatistic.  The only difference seems to be that setStatistic also enables the statistic.  Does it make sense to call the create code in the set function so that the creation code isn't redundant and will be easier to maintain should we chance some of the initialization in the future?